lib/Language/Haskell/Stylish/Parse.hs

```diff
@@ -44,11 +44,11 @@ unCpp = unlines . go False . lines
 
 
 --------------------------------------------------------------------------------
--- | Remove shebang from the first line
+-- | Remove shebang lines
 unShebang :: String -> String
-unShebang str
-    | "#!" `isPrefixOf` str = unlines $ ("" :) $ drop 1 $ lines str
-    | otherwise             = str
+unShebang str =
+    let (shebangs, other) = break (not . ("#!" `isPrefixOf`)) (lines str) in
+    unlines $ map (const "") shebangs ++ other
```

Comment:
I just realised this could also be `unlines $ ("" <$ shebangs) ++ other` and I can change it to that if you'd prefer.